homeassistant/components/simplisafe/__init__.py

```diff
@@ -231,9 +243,33 @@ def _async_register_base_station(
     )
 
 
-async def async_setup_entry(  # noqa: C901
-    hass: HomeAssistant, entry: ConfigEntry
-) -> bool:
+@callback
+def _async_get_system_for_service_call(
+    hass: HomeAssistant, call: ServiceCall
+) -> SystemV2 | SystemV3:
+    """Get the SimpliSafe system related to a service call (by device ID)."""
+    device_id = call.data[ATTR_DEVICE_ID]
+    device_registry = dr.async_get(hass)
+
+    if alarm_control_panel_device_entry := device_registry.async_get(device_id):
+        if TYPE_CHECKING:
+            assert alarm_control_panel_device_entry.via_device_id
+        if base_station_device_entry := device_registry.async_get(
+            alarm_control_panel_device_entry.via_device_id
+        ):
+            (identifiers,) = base_station_device_entry.identifiers
+            [system_id] = [i for i in identifiers if i != DOMAIN]
+            for entry in hass.config_entries.async_entries(DOMAIN):
+                if entry.entry_id in alarm_control_panel_device_entry.config_entries:
```

Comment:
Why not do it the other way around, and iterate over `base_station_device_entry.config_entries` ?